src/Extensions/AzureBlobPayloads/Interceptors/BasePayloadInterceptor.cs

```diff
@@ -0,0 +1,225 @@
+                    {
+                        t.Result.Dispose();
+                    }
+                },
+                CancellationToken.None,
+                TaskContinuationOptions.ExecuteSynchronously,
+                TaskScheduler.Default);
+        }
+
+        return new AsyncUnaryCall<TResponse>(
+            ResponseAsync(),
+            ResponseHeadersAsync(),
+            GetStatus,
+            GetTrailers,
+            Dispose);
+    }
+
+    // Server streaming: resolve payloads in streamed responses (e.g., GetWorkItems)
+
+    /// <inheritdoc/>
+    public override AsyncServerStreamingCall<TResponse> AsyncServerStreamingCall<TRequest, TResponse>(
+        TRequest request,
+        ClientInterceptorContext<TRequest, TResponse> context,
+        AsyncServerStreamingCallContinuation<TRequest, TResponse> continuation)
+    {
+        // For streaming, request externalization is not needed currently
+        AsyncServerStreamingCall<TResponse> call = continuation(request, context);
+
+        IAsyncStreamReader<TResponse> wrapped = new TransformingStreamReader<TResponse>(call.ResponseStream, async (msg, ct) =>
+        {
+            await this.ResolveResponsePayloadsAsync(msg, ct);
+            return msg;
+        });
+
+        return new AsyncServerStreamingCall<TResponse>(
+            wrapped,
+            call.ResponseHeadersAsync,
+            call.GetStatus,
+            call.GetTrailers,
+            call.Dispose);
+    }
+
+    /// <summary>
+    /// Externalizes large payloads in request messages.
+    /// </summary>
+    /// <typeparam name="TRequest">The request type.</typeparam>
+    /// <param name="request">The request to process.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected abstract Task ExternalizeRequestPayloadsAsync<TRequest>(TRequest request, CancellationToken cancellation);
+
+    /// <summary>
+    /// Resolves payload tokens in response messages.
+    /// </summary>
+    /// <typeparam name="TResponse">The response type.</typeparam>
+    /// <param name="response">The response to process.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected abstract Task ResolveResponsePayloadsAsync<TResponse>(TResponse response, CancellationToken cancellation);
+
+    /// <summary>
+    /// Externalizes a payload if it exceeds the threshold.
+    /// </summary>
+    /// <param name="assign">Action to assign the externalized token.</param>
+    /// <param name="value">The value to potentially externalize.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected Task MaybeExternalizeAsync(Action<string?> assign, string? value, CancellationToken cancellation)
```

Comment:
This is a kind of weird interface. We take a function to assign the value back to something? It's fairly opaque what this is doing and I wonder if it's better to return the string in the `Task` and let the caller assign it. The end goal is the same but, to me, the control feels a bit inverted this way.